parts/k8s/kubernetesagentresourcesvmas.t

```diff
@@ -5,20 +5,17 @@
         "name": "loop"
       },
       "dependsOn": [
-{{if .IsCustomVNET}}
-      "[variables('nsgID')]"
-{{else}}
-      "[variables('vnetID')]"
+{{if not .IsCustomVNET}}
```

Comment:
To summarize in English.

Before: if custom VNET depend on nsgID / else depend on vnetID

After: if custom VNET depend on nsgID / else depend on vnetID + nsgID